src/domain/course/CoursePageContainer.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLocation, useParams } from 'react-router';
+import { Link } from 'react-router-dom';
+
+import ErrorHero from '../../common/components/error/ErrorHero';
+import LoadingSpinner from '../../common/components/spinner/LoadingSpinner';
+import { useCourseDetailsQuery } from '../../generated/graphql';
+import useLocale from '../../hooks/useLocale';
+import isClient from '../../util/isClient';
+import MainContent from '../app/layout/MainContent';
+import PageWrapper from '../app/layout/PageWrapper';
+import { ROUTES } from '../app/routes/constants';
+import EventClosedHero from '../event/eventClosedHero/EventClosedHero';
+import EventContent from '../event/eventContent/EventContent';
+import EventHero from '../event/eventHero/EventHero';
+import EventPageMeta from '../event/eventPageMeta/EventPageMeta';
+import { isEventClosed } from '../event/EventUtils';
+import { useSimilarCoursesQuery } from '../event/queryUtils';
+import SimilarEvents from '../event/similarEvents/SimilarEvents';
+import { EventFields, EventType } from '../event/types';
+import styles from './coursePage.module.scss';
+
+interface RouteParams {
+  id: string;
+}
+
+const CoursePageContainer: React.FC = () => {
+  const { t } = useTranslation();
+  const { search } = useLocation();
+  const params = useParams<RouteParams>();
+  const courseId = params.id;
+  const locale = useLocale();
+
+  const { data: courseData, loading } = useCourseDetailsQuery({
+    variables: {
+      id: courseId,
+      include: ['in_language', 'keywords', 'location', 'audience'],
+    },
+  });
+
+  const course = courseData?.courseDetails;
+
+  const courseClosed = !course || isEventClosed(course);
+
+  return (
+    <PageWrapper className={styles.coursePageWrapper} title="event.title">
+      <MainContent offset={-70}>
+        <LoadingSpinner isLoading={loading}>
+          {course ? (
+            <>
+              {/* Wait for data to be accessible before updating metadata */}
+              <EventPageMeta event={course} />
+              {courseClosed ? (
```

Comment:
This could be simpler?
```suggestion
             {courseClosed ? (
                <EventClosedHero />
              ) : (
                <EventHero event={course} eventType={EventType.COURSE} />
                <EventContent event={course} eventType={EventType.COURSE} />
              )}
```

Reply:
hmm CoursePageContainer and EventPageContainer should be maybe combined to use same codem like I just did for the searchPage

because this jsx part is copy paste from EventPageContainer

Reply:
But you are right, your suggestion is better, good catch

Reply:
> hmm CoursePageContainer and EventPageContainer should be maybe combined to use same codem like I just did for the searchPage

Yes I was about to suggest that but thinked that you mentioned previously that some of the refactoring will be left later or so... Anyway combining these two would be great.